sklearn/ensemble/_hist_gradient_boosting/common.pyx

```diff
@@ -10,6 +10,11 @@ X_BINNED_DTYPE = np.uint8  # hence max_bins == 256
 G_H_DTYPE = np.float32
 X_BITSET_INNER_DTYPE = np.uint32
 
+# Note that we use Y_DTYPE=float64 to avoid issues with floating point precision when
+# summing gradients and hessians. Those are difficult to protect via tools like
```

Comment:
Are these links actionable or purely informative? I'm +0 on referring to different summation methods if there are purely informative.

Reply:
Python 3.12 uses Neumaier summation and numpy uses pairwise summation, both to protect for loss of floating point precision. Should I
1. Remove the links
2. Remove the summation names
3. Remove both

Reply:
I am okay with including the links if we make it explicit that these summation methods are possible paths to use `float32` for `sum_gradients`.

Although, I do not think we could use `float32` anyways because of overflowing. (`float64` has a much higher max value).

Reply:
Note that the first item discussed in https://github.com/hcho3/xgboost-fast-hist-perf-lab/issues/15 is using float32 for histograms instead of float64.